evalml/automl/automl_algorithm/evalml_algorithm.py

```diff
@@ -0,0 +1,365 @@
+    def _create_naive_pipelines_with_feature_selection(self):
+        feature_selector = (
+            RFRegressorSelectFromModel
+            if is_regression(self.problem_type)
+            else RFClassifierSelectFromModel
+        )
+        estimators = self._naive_estimators()
+        pipelines = [
+            make_pipeline(
+                self.X,
+                self.y,
+                estimator,
+                self.problem_type,
+                sampler_name=self._sampler_name,
+                extra_components=[feature_selector],
+            )
+            for estimator in estimators
+        ]
+        return pipelines
+
+    def _create_tuner(self, pipeline):
+        pipeline_hyperparameters = pipeline.get_hyperparameter_ranges(
+            self._custom_hyperparameters
+        )
+        self._tuners[pipeline.name] = self._tuner_class(
+            pipeline_hyperparameters, random_seed=self.random_seed
+        )
+
+    def _create_fast_final(self):
+        estimators = [
+            estimator
+            for estimator in get_estimators(self.problem_type)
+            if estimator not in self._naive_estimators()
+        ]
+        pipelines = [
+            make_pipeline(
+                self.X,
+                self.y,
+                estimator,
+                self.problem_type,
+                sampler_name=self._sampler_name,
+                extra_components=[SelectColumns],
+            )
+            for estimator in estimators
+        ]
+        pipelines = [
+            pipeline.new(
+                parameters={
+                    "Select Columns Transformer": {"columns": self._selected_cols}
+                },
+                random_seed=self.random_seed,
+            )
+            for pipeline in pipelines
+        ]
+
+        for pipeline in pipelines:
+            self._create_tuner(pipeline)
+        return pipelines
+
+    def _create_ensemble(self):
+        input_pipelines = []
+        for pipeline_dict in self._best_pipeline_info.values():
+            pipeline = pipeline_dict["pipeline"]
+            pipeline_params = pipeline_dict["parameters"]
+            parameters = self._transform_parameters(pipeline, pipeline_params)
+            input_pipelines.append(
+                pipeline.new(parameters=parameters, random_seed=self.random_seed)
+            )
+        n_jobs_ensemble = 1 if self.text_in_ensembling else self.n_jobs
+        ensemble = _make_stacked_ensemble_pipeline(
+            input_pipelines,
+            input_pipelines[0].problem_type,
+            random_seed=self.random_seed,
+            n_jobs=n_jobs_ensemble,
+        )
+        return [ensemble]
+
+    def _create_long_top_n(self, n):
+        estimators = [
+            (pipeline_dict["pipeline"].estimator, pipeline_dict["mean_cv_score"])
+            for pipeline_dict in self._best_pipeline_info.values()
+        ]
+        estimators.sort(key=lambda pipeline: pipeline[1])
+        estimators = estimators[:n]
```

Comment:
I see that this is choosing the best estimators based on the `mean_cv_score`, which makes sense. I think the sorting should be done based on what objective is being used, since `R2==1` is the best for regression problems but `log loss == 0` is the best for classification problems.

Reply:
all the scores coming in should have been [converted to "lower is better"](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_search.py#L1129) so this order should be ok!